File: src/backend/src/main/java/br/com/grupopipa/gestaointegrada/atendimento/agendamento/agendaregra/dto/AgendaRegraDTO.java
package br.com.grupopipa.gestaointegrada.atendimento.agendamento.agendaregra.dto;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;
import java.util.List;
import java.util.Set;
import java.util.UUID;

import com.fasterxml.jackson.annotation.JsonFormat;

import br.com.grupopipa.gestaointegrada.core.dto.DTO;
import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class AgendaRegraDTO implements DTO {

    private UUID id;
    private UUID agendaId;

    @JsonFormat(pattern = "yyyy-MM-dd")
    private LocalDate dataInicio;

    @JsonFormat(pattern = "yyyy-MM-dd")
    private LocalDate dataFim;

    @JsonFormat(pattern = "HH:mm")
    private LocalTime horaInicio;

    @JsonFormat(pattern = "HH:mm")
    private LocalTime horaFim;

    private Integer duracaoSessaoMinutos;
    private Set<String> diasSemana;
    private Set<UUID> convenioIds;
    private List<String> convenioNomes;
    private Set<UUID> procedimentoIds;
    private List<String> procedimentoNomes;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private String createdBy;
    private String updatedBy;
}

File: src/frontend/src/app/components/atendimento/agendamento/agendaregra/model/agenda-regra-dto.ts
import { Exclude, Expose } from 'class-transformer';

@Exclude()
export class AgendaRegraDTO {
  @Expose() id?: string;
  @Expose() agendaId = '';
  @Expose() dataInicio = '';
  @Expose() dataFim?: string;
  @Expose() horaInicio = '';
  @Expose() horaFim = '';
  @Expose() duracaoSessaoMinutos = 30;
  @Expose() diasSemana: string[] = [];
  @Expose() convenioIds: string[] = [];
  @Expose() convenioNomes?: string[];
  @Expose() procedimentoIds: string[] = [];
  @Expose() procedimentoNomes?: string[];
}
